e2e-tests/tests/infra/StudioPlaywrightPage.ts

```diff
@@ -233,4 +233,35 @@ export default class StudioPlaywrightPage {
   getComponentPath(componentName: string) {
     return path.join(this.tmpDir, "src/components", componentName + ".tsx");
   }
+
+  async waitForIFrameImagesToLoad() {
+    const images = await this.preview.getByRole("img").all();
+    const imgPromises: Promise<void>[] = [];
+
+    for (const img of images) {
+      const pollForImageReady = expect
+        .poll(
+          async () => {
+            return await img.evaluate((e) => (e as HTMLImageElement).complete);
+          },
+          {
+            message: "Wait for images in page preview to render.",
+            timeout: 1000,
+          }
+        )
+        .toBeTruthy();
+      imgPromises.push(pollForImageReady);
+    }
+    await Promise.all(imgPromises);
```

Comment:
what you have def works! but you can use .map to simplify a little
```ts
const imgPromises = images.map(i => {
  expect.poll(() => img.evaluate(blahblahbalb))
})
await Promise.all(imgPromises
```

Reply:
🧠